### PR TITLE
operation log: add library name after operator

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Closes rero/rero-ils#2059.

### Dependencies
- https://github.com/rero/rero-ils/pull/3311

### CODE REVIEW

**Review by zannkukai — COMMENTED** [review]
How are you sure than `metadata.library` is the selected switchLibrary entry at the moment of the operation ? 

As an exemple: 
* logged as Astrid working on AOSTE1
* Edit an item of AOASTE1, switch location to "AOSTE2.any_location"
* The opLog.metadata.library will be "AOSTE2", but the operation was made by "Astrid logged on AOSTE1"

As the switchLibrary entry is not a value known by the backend, it can't be exploited.
For me the original issue cannot be solved in current system implementation

**Comment by PascalRepond** [comment]
> As the switchLibrary entry is not a value known by the backend, it can't be exploited. For me the original issue cannot be solved in current system implementation

Exactly, the switch library is known only by the frontend. This causes multiple problems, also for stats of created resources because we don't know the library for librarians with multiple libraries. However, this implementation should be acceptable at the moment as librarians with multiple libraries are an exception. But we should think about how to make it cleaner.

**Review by PascalRepond — CHANGES_REQUESTED** [review]
- [x] Displaying document operation logs containing librarians from another organisation results in 403 errors because of permissions

**Comment by zannkukai** [comment]
> However, this implementation should be acceptable at the moment as librarians with multiple libraries are an exception. But we should think about how to make it cleaner.

We know this solution isn't a complete and valid solution but it's correct : Work now, and in some weeks work again ... 
Or maybe try to find a complete solution (and more time consuming) now and take a coffee in some weeks.
As it seems not a critical issue at this time, my opinion is to find a complete solution now.

**Comment by PascalRepond** [comment]
> We know this solution isn't a complete and valid solution but it's correct : Work now, and in some weeks work again ... Or maybe try to find a complete solution (and more time consuming) now and take a coffee in some weeks. As it seems not a critical issue at this time, my opinion is to find a complete solution now.

This would be scope creep in my opinion. This issue addresses a small and specific demand. Although a larger problem (`patron.library`: https://github.com/rero/rero-ils/issues/3310) has an influence on this, this doesn't mean that we have to fix everything at the same time. Issue 3310 should be addressed in time when it has been more precisely specified.

**Review by PascalRepond — APPROVED** [review]

File: projects/admin/src/app/record/operation-logs/operation-logs.component.html [hunk 1]
```diff
@@ -58,6 +58,9 @@ <h5 class="modal-title" translate>Operation history</h5>
       {{ metadata.date | dateTranslate :'shortDate' }} - {{ metadata.date | dateTranslate :'mediumTime' }}
     </div>
     <div class="col-2 operation">{{ metadata.operation | translate }}</div>
-    <div class="col">{{ metadata.user_name }}</div>
+    <div class="col">
+      {{ metadata.user_name }}
+      <ng-container *ngIf="metadata.library">- {{ metadata.library.name }}</ng-container>
```
Comment (lauren-d) [inline projects/admin/src/app/record/operation-logs/operation-logs.component.html:63]: COSMETIC: Is there a missing space before the hyphen ? 
`- {{ metadata.library.name }}` or ` - {{ metadata.library.name }}`

**Review by lauren-d — APPROVED** [review]

**Comment by ManaDeweerdt** [comment]
- [ ] In case of Astrid with multiple login location, the operation log will always be 'Bibliothèque de la Vallée d'AOSTE', no matter what login location. It should be the login location.
image

**Review by benerken — CHANGES_REQUESTED** [review]
Thanks for this PR but it will create another issue for users with multiple libraries. Sorry but it's not the solution to solve an issue for one kind of users and let down the others.

**Pull request merged**